app/components/projects-search-selector.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { Component, PropTypes } from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import Select from 'react-select';
+import debounce from 'debounce';
+
+class SearchSelector extends Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.navigateToProject = this.navigateToProject.bind(this);
+    this.searchByName = this.searchByName.bind(this);
+  }
+
+  navigateToProject(projectId) {
+    apiClient.type('projects').get(projectId)
+      .then(project => {
+        if (project.redirect != null && project.redirect.length !== 0) {
+          return window.location.href = project.redirect;
+        }
+        return window.location.href = ['/projects', project.slug].join('/');
```

Comment:
This isn't the best way to navigate around - the router object should be available on the context, why not use `push`?